pyvelox/pyvelox.h

```diff
@@ -168,4 +257,43 @@ inline void addVeloxBindings(pybind11::module& m, bool asLocalModule = true) {
   rowType.def("names", &RowType::names, "Return the names of the columns");
 }
 
+inline void addVectorBindings(
+    pybind11::module& m,
+    bool asModuleLocalDefinitions = true) {
+  namespace py = pybind11;
+  std::shared_ptr<facebook::velox::memory::MemoryPool> pool =
+      facebook::velox::memory::getDefaultMemoryPool();
```

Comment:
I am also very curious on how this contrasts with pythons memory management. 
Say i do something like a = flatVector (1, 2, 3) and a goes out of scope then I suppose we also end up freeing buffers associated with the vector .

Reply:
Yes, `pybind11` just holds the `shared_ptr` to the vector, so when `a` goes out of scope it will free it. If you're asking about the MemoryPool, the reference is being held by the lambda in `from_list`. This memory pool will end up getting freed when the class object gets freed, which happens when the module is freed. An alternative spot to put the memory pool could be as a property of the module itself (so it would be a module-global memory pool).